compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -562,9 +564,10 @@ public boolean isArrayOpenSealedType(BArrayType arrayType) {
      */
     private SymbolEnv findEnclosingInvokableEnv(SymbolEnv env, BLangInvokableNode encInvokable) {
         if (env.enclEnv.node != null && env.enclEnv.node.getKind() == NodeKind.ARROW_EXPR) {
+            // if enclosing env's node is arrow expression
             return env.enclEnv;
         }
-        if (env.enclInvokable == encInvokable) {
+        if (env.enclInvokable == encInvokable && env.enclInvokable != null) {
```

Comment:
Is this check correct? We are doing a null check after checking for an equivalency on the same?

Reply:
In arrow expression the `encInvokable` also becomes null. Makes sense to do the null check first though. Fixed in commit 321ba02